Akshaya branch

akshayaramaswamy opened the pull request:
Sending this for review

akshayaramaswamy committed:
Interacting with the API and extracting the leads.

akshayaramaswamy committed:
Updating the client to pre-fill name and email from the API.

aa920502 commented on client/package.json:
```diff
@@ -13,7 +13,7 @@
     "react-router": "^6.3.0",
     "react-router-dom": "^6.3.0",
-    "react-scripts": "5.0.1",
+    "react-scripts": "^2.1.3",
```
why are you using an older version here?

aa920502 commented on client/src/components/RegistrationFormWithContinuedFlow.js:
```diff
@@ -15,7 +14,19 @@ function RegistrationFormWithContinuedFlow() {
   });
 
-  const addFormHandler = (event) => {
+  const [backendData, setBackendData] = useState([{}]);
```
why do you need to make this call here?

akshayaramaswamy replied:
The axios.get() returns a promise variable, made this call to consume the data from the promise variable.

aa920502 replied:
The call itself is fine, let's update the endpoint, and initiate a POST request from here to the backend, and update the route handling logic in routes.js  in backend correspondingly

aa920502 commented on client/src/components/RegistrationFormWithContinuedFlow.js:
```diff
@@ -75,11 +85,11 @@ function RegistrationFormWithContinuedFlow() {
             className={styles.formControl}
             name="name"
-            value={addFormData.name}
+            value={nameVar}
```
can we using a better naming here?

aa920502 commented on server/routes/routes.js:
```diff
@@ -1,9 +1,14 @@
 const signUpTemplateCopy = require("../models/SignUpModels");
+const server_value = require("../server")
 
 router.get("/", (req, res) => {
```
can we use another endpoint? something like /lead_retrieval?

akshayaramaswamy committed:
Addressing code comments.

akshayaramaswamy committed:
Addressing code comments.

aa920502 commented on server/server.js:
```diff
@@ -12,8 +15,43 @@ var corsOptions = {
   optionsSuccessStatus: 200, // some legacy browsers (IE11, various SmartTVs) choke on 204
 };
 
+const api = bizSdk.FacebookAdsApi.init(process.env.ACCESS_TOKEN);
```
Let's move this logic into routes.js, also let's remove console.logs when you push the code

aa920502 commented on server/server.js:
```diff
@@ -12,8 +15,43 @@ var corsOptions = {
+
+let sample_code, field_data, response;
+module.exports.getRawValue = async function getRawValue(){
+  sample_code = await (new Lead(process.env.LEAD_ID)).get(
```
You should scrape the lead ID from the URL and pass it from frontend, instead of reading from env file

akshayaramaswamy committed:
Fixing merge conflicts.

Merge branch 'akshaya_branch' of https://github.com/aa920502/continued-flow-sample-site into akshaya_branch

aa920502 commented on client/src/components/RegistrationFormWithContinuedFlow.js:
```diff
@@ -16,14 +17,23 @@ function RegistrationFormWithContinuedFlow() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const fb_lead_id = searchParams.get('fbld_id');
+
+  axios.post('/url', {url: fb_lead_id})
```
I think {lead_id: fb_lead_id} is much more clear than url, since this is not the url value
